### PR TITLE
Add non-interactive login to bakery.

### PR DESCRIPTION
**Pull request by jaycee**

### CODE REVIEW

**Comment by jaycee** [comment]
QA:
This requires an environment. You can setup an environment of your own if you choose--ping if you want to do that. There is also an environment already up running this code. The url for openstack dashboard there is https://54.81.141.241/horizon. If you use this environment you'll need to allow mixed-content as the idm and jem are not configured for https in this environment.

Go to openstack dashboard and login. Open the GUI and open your developer console. Once the GUI finishes loading you should see the message "Environment listing success" and a log of an empty array. There are no environments in this system for the openstack user at this time. You can also use the network panel of the developer tools to ensure you're seeing requests against the jem at /v1/env and the idm at both /login and /login/idp/jujugui

File: jujugui/static/gui/src/test/test_bakery.js [hunk 1]
```diff
@@ -18,7 +18,7 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 'use strict';
 
-describe('Bakery', function() {
+describe.only('Bakery', function() {
```
Comment (makyo) [inline jujugui/static/gui/src/test/test_bakery.js]: Remove .only

**Comment by makyo** [comment]
:+1: QA okay!

**Comment by bac** [comment]
Code :+1:

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3156/
Test PASSed.

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3158/
Test FAILed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js [hunk 2]
```diff
@@ -76,16 +76,16 @@ var module = module;
...
+   *     e.g. rose/tardis.
    * @param success {function} A callback to be called on success. Should
    *   take an object with environment data as its one parameter.
    * @param failure {function} A callback to be called on failure. Should
    *   take an error message as its one parameter.
    */
-  environment.prototype.getEnvironment = function (
-      username, envName, success, failure) {
-    var url = [this.jemUrl, 'env', username, envName].join('/');
+  environment.prototype.getEnvironment = function (envName, success, failure) {
```
Comment (hatched) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: Typically in a nodejs library a function is passed a single callback whos job it is to handle both success and failures. It's passed the error as the first parameter and then any values afterwards. So in interest of sticking with that flow I'd like to update these methods to take just a single callback for both outcomes.

File: jujugui/static/gui/src/test/test_bakery.js [hunk 3]
```diff
@@ -43,10 +42,28 @@ describe('Bakery', function() {
     bakery = null;
   });
 
-  it('can be instantiated with the proper config values', function() {
-    assert.equal(
-        bakery.webhandler instanceof Y.juju.environments.web.WebHandler,
-        true);
+   it('can be instantiated with the proper config values', function() {
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_bakery.js]: indent

File: jujugui/static/gui/src/test/test_bakery.js (hunk 1 above)
Comment (jaycee) [inline jujugui/static/gui/src/test/test_bakery.js]: Good catch, thanks.

File: jujugui/static/gui/src/test/test_bakery.js (hunk 3 above)
Comment (jaycee) [inline jujugui/static/gui/src/test/test_bakery.js]: Fixed.

File: jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js (hunk 2 above)
Comment (jaycee) [inline jujugui/static/gui/src/app/assets/javascripts/jujulib/juju.js]: Seems reasonable, but out of scope for this branch given necessary test updating etc. I'll do it as a driveby in the next juju.js branch I'll be working on.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3165/
Test PASSed.

**Comment by jaycee** [comment]
Thanks for the reviews @bac  and @makyo. :shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/3166/
Test PASSed.

**Pull request merged**